### ISSUE TITLE
suspend-then-hibernate doesn't fail safe (suspend) if hibernation fails

### ISSUE THREAD

**Issue by smcv** [systemd/systemd#10212]
**systemd version the issue has been seen with**

239-10 (Debian)

**Used distribution**

Debian unstable

**Steps to reproduce the problem**

* Use a lot of RAM + swap, enough that hibernation will fail
* Suspend using `suspend-then-hibernate` (the default in recent gnome-settings-daemon versions, although I think that's an unwise default and I've reported it as a g-s-d bug)

**Expected behaviour you didn't see**

* Laptop wakes up from suspend-to-RAM
* Laptop tries to hibernate (suspend to disk)
* Hibernation fails
* Laptop returns to suspend-to-RAM, using minimal battery power

**Unexpected behaviour you saw**

* Laptop wakes up from suspend-to-RAM
* Laptop tries to hibernate (suspend to disk)
* Hibernation fails
* Laptop continues to run normally

As well as consuming more battery power, if the laptop is in a poorly ventilated location (for example being carried in a backpack) it could result in overheating, and if the laptop uses a mechanical hard disk it could result in damage to the disk.

**Comment by smcv** [comment]
Example log messages:

```
Sep 30 01:15:32 espresso systemd-sleep[12098]: Failed to write /sys/power/state: No space left on device
Sep 30 01:15:32 espresso kernel: PM: hibernation exit
Sep 30 01:15:32 espresso systemd[1]: systemd-suspend-then-hibernate.service: Main process exited, code=exited, status=1/FAILURE
Sep 30 01:15:32 espresso systemd[1]: systemd-suspend-then-hibernate.service: Failed with result 'exit-code'.
Sep 30 01:15:32 espresso systemd[1]: Failed to start Suspend; Hibernate if not used for a period of time.
Sep 30 01:15:32 espresso systemd[1]: Dependency failed for Suspend; Hibernate if not used for a period of time.
Sep 30 01:15:32 espresso systemd[1]: suspend-then-hibernate.target: Job suspend-then-hibernate.target/start failed with result 'dependency'.
Sep 30 01:15:32 espresso systemd[1]: sleep.target: Unit not needed anymore. Stopping.
```

**Comment by poettering** [comment]
Yeah, indeed we should log a recognizable log message in that case, and go back to sleep.

**Comment by poettering** [comment]
Fix waiting in #10504
